webviz_subsurface/plugins/_volumetric_analysis/controllers/comparison_controllers.py

```diff
@@ -479,3 +492,80 @@ def add_fluid_zone_column(dframe: pd.DataFrame, filters: dict) -> pd.DataFrame:
     if "FLUID_ZONE" not in dframe and "FLUID_ZONE" in filters:
         dframe["FLUID_ZONE"] = (" + ").join(filters["FLUID_ZONE"])
     return dframe
+
+
+def get_selected_values(selections: dict) -> Tuple[str, str]:
+    if not "|" in selections["value1"]:
+        return selections["value1"], selections["value2"]
+    ens1, sens1 = selections["value1"].split("|")
+    ens2, sens2 = selections["value2"].split("|")
+    return (sens1, sens2) if ens1 == ens2 else (ens1, ens2)
+
+
+def create_waterfall_figures(
+    df: pd.DataFrame, selections: dict, groups: List[str], max_figures: int
+) -> List[go.Figure]:
+    """
+    Create Water Fall plots showing volume change contributions, using
+    the comparison table as input. A maximum number of plots has been set
+    to reduce computation time if e.g. REAL is used in the groups.
+    The HC volume formula is as follows
+    Volume = (GRV*NTG*PORO*(1-SW)) / formation volume factor (Bo/Bg)
+    For properties that are numerators in this formula their diff in %
+    can be used to determine volume impact. Bo/BG is handled slightly
+    different as it is the denominator.
+    """
+
+    response = selections["Response"]
+    val1, val2 = get_selected_values(selections)
+
+    sat_col = "SO" if response == "STOIIP" else "SG"
+    fvf_col = "BO" if response == "STOIIP" else "BG"
```

Comment:
Maybe more future proof to verify GIIP for SG/BG and raise a `NotImplementedError` if neither STOIIP or GIIP.